ex_cubic_ods_ingestion/lib/ex_cubic_ods_ingestion/repo/migrator.ex

```diff
@@ -0,0 +1,45 @@
+defmodule ExCubicOdsIngestion.Repo.Migrator do
+  @moduledoc """
+  GenServer which runs on startup to run Ecto migrations, then terminates.
+  """
+  require Logger
+
+  @opts [module: Ecto.Migrator, run_migrations_at_startup?: true]
+
+  @spec start_link(Keyword.t()) :: :ignore
+  def start_link(opts) do
+    opts = Keyword.merge(@opts, opts)
+
+    if Keyword.get(opts, :run_migrations_at_startup?) do
+      _ignored = Logger.info("Migrating synchronously")
+      migrate!(opts[:module])
+      _ignored = Logger.info("Finished migrations")
+    end
+
+    :ignore
+  end
+
+  @spec child_spec(Keyword.t()) :: Supervisor.child_spec()
+  def child_spec(opts) do
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, [opts]},
+      restart: :transient
+    }
+  end
+
+  defp migrate!(module) do
+    for repo <- [ExCubicOdsIngestion.Repo] do
+      _ignored = Logger.info(fn -> "Migrating repo=#{repo}" end)
+
+      {time_usec, {:ok, _, _}} =
+        :timer.tc(module, :with_repo, [repo, &module.run(&1, :up, all: true)])
+
+      time_msec = System.convert_time_unit(time_usec, :microsecond, :millisecond)
+      _ignored = Logger.info(fn -> "Migration finished repo=#{repo} time=#{time_msec}" end)
```

Comment:
suggestion: with more recent versions of Elixir, you don't need either the `_ignored` or the function:
```suggestion
      Logger.info("Migration finished repo=#{repo} time=#{time_msec}")
```